### DISCUSSION TITLE
Enable/Disable Vue Devtools at runtime

### DISCUSSION THREAD

**Discussion by mlpierce22** [nuxt/nuxt#19546] [Questions]
I know that I can enable or disable vue devtools by setting the `process.env.NODE_ENV` to `development`/`production` on build or run. I've done a fair amount of research, but I can't for the life of me figure out if there is any way to enable or disable devtools on the fly (overriding the build/run settings.)

I've tried setting the `process.env.NODE_ENV` value via middleware, but this doesn't work because nuxt replaces that variable with it's string equivalent. I've also tried using pinia, but I can't figure out what values devtools uses. I know there is: `__VUE_PROD_DEVTOOLS__` and `__VUE_DEVTOOLS_GLOBAL_HOOK__` but they seem like they still need to be set before the page is initialized. 

For example, when devtools is disabled, I have these globally accessible in the window.
<img width="621" alt="image" src="https://user-images.githubusercontent.com/35623597/223912130-aa6e2026-20c4-40e0-9250-75a1f98000fb.png">

And when devtools is enabled, I have these
<img width="804" alt="image" src="https://user-images.githubusercontent.com/35623597/223912294-d5fa82be-4402-4399-b206-1e27001a05af.png">

I've also tried installing the standalone thing to my app like mentioned [here](https://devtools.vuejs.org/guide/installation.html#using-dependency-package) but haven't been able to get that to work, and unfortunately the Nuxt3 devtools are still super early and don't seem like they have any way to turn them on/off dynamically.

Basically, what I'm trying to do is have something like a query parameter like `?devtools=enable`, that I could check for, and then enable devtools depending on if it is present, regardless of whether or not it has been built for production.

I'm open to creative/all solutions! Thanks!

related unanswered questions: https://github.com/nuxt/nuxt/discussions/16237, https://github.com/nuxt/nuxt/discussions/18293
